devices/AtModem/Http/Http/Uri.cs

```diff
@@ -0,0 +1,1210 @@
+        /// <p>The <paramref name="uriString"/> is empty.</p>
+        /// <p>-or-</p><p>The scheme specified in <paramref name="uriString"/>
+        /// is not correctly formed.  </p>
+        /// <p>-or-</p><p><paramref name="uriString"/> contains too many
+        /// slashes.</p>
+        /// <p>-or-</p><p>The password specified in <paramref name="uriString"/>
+        /// is not valid.</p>
+        /// <p>-or-</p><p>The host name specified in
+        /// <paramref name="uriString"/> is not valid.</p>
+        /// <p>-or-</p><p>The file name specified in
+        /// <paramref name="uriString"/> is not valid.</p>
+        /// <p>-or-</p><p>The user name specified in
+        /// <paramref name="uriString"/> is not valid.</p>
+        /// <p>-or-</p><p>The host or authority name specified in
+        /// <paramref name="uriString"/> cannot be terminated by backslashes.
+        /// </p>
+        /// <p>-or-</p><p>The port number specified in
+        /// <paramref name="uriString"/> is not valid or cannot be parsed.</p>
+        /// <p>-or-</p><p>The length of <paramref name="uriString"/> exceeds
+        /// 65534 characters.</p>
+        /// <p>-or-</p><p>The length of the scheme specified in
+        /// <paramref name="uriString"/> exceeds 1023 characters.</p>
+        /// <p>-or-</p><p>There is an invalid character sequence in
+        /// <paramref name="uriString"/>.</p>
+        /// <p>-or-</p><p>The MS-DOS path specified in
+        /// <paramref name="uriString"/> must start with c:\\.</p>
+        /// </exception>
+        public Uri(string uriString)
+        {
+            if (uriString is null)
+            {
+                throw new ArgumentNullException();
+            }
+
+            if (!ConstructAbsoluteUri(uriString))
+            {
+                throw new ArgumentException();
+            }
+        }
+
+        /// <summary>
+        /// Constructs an absolute Uri from a URI string.
+        /// </summary>
+        /// <param name="uriString">A URI.</param>
+        /// <remarks>
+        /// See <see cref="Uri(string)"/>.
+        /// </remarks>
+        protected bool ConstructAbsoluteUri(string uriString)
+        {
+            // ParseUriString provides full validation including testing for null.
+            if (TryParseUriString(uriString))
+            {
+                _OriginalUriString = uriString;
+
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Constructs Uri from string and enumeration that tell what is the type of Uri.
+        /// </summary>
+        /// <param name="uriString">String to construct Uri from</param>
+        /// <param name="kind">Type of Uri to construct</param>
+        /// <exception cref="ArgumentException">The scheme specified in the URI formed by combining <paramref name="baseUri"/> and <paramref name="relativeUri"/> is not valid.</exception>
```

Comment:
This method doesn't have any `ArgumentException` rather several `FormatException`...